src/Storage/Storage/Blob/Cmdlet/NewAzureStorageBlobSasToken.cs

```diff
@@ -176,7 +182,9 @@ public override void ExecuteCmdlet()
 
             // When the input context is Oauth bases, can't generate normal SAS, but UserDelegationSas
             bool generateUserDelegationSas = false;
-            if (Channel != null && Channel.StorageContext != null && Channel.StorageContext.StorageAccount.Credentials != null && Channel.StorageContext.StorageAccount.Credentials.IsToken)
+            if (Channel != null && Channel.StorageContext != null && (
+                (Channel.StorageContext.StorageAccount.Credentials != null && Channel.StorageContext.StorageAccount.Credentials.IsToken)
+                || (Channel.StorageContext.Track2OauthToken != null)))
```

Comment:
Should throw error if parameter DelegatedUserObjectId is not null, but user not use oauth authentication.